### ISSUE TITLE
Wiki-style markup for labels

### ISSUE THREAD

**Issue by drom** [drom/wavedrom#67]
_From [alex.d...@gmail.com](https://code.google.com/u/112117906073099522199/) on March 01, 2013 16:03:16_

Just some ideas for enhancement:

Do we need to allow Google wiki-style markup for labels (signal names, timing, phase labels)?

In order of importance:

type              wiki-markup         use case
----              -----------         --------
subscript         ,,sub,,script       useful for timing labels t1, t2 ...
overline          ???                 for negative signal polarity

`code`
*bold*
_italic_         - we have to be careful about normal usage of underscore in names 
^super^script    - may be messed up with XOR?
~~strikeout~~
underline?

_Original issue: http://code.google.com/p/wavedrom/issues/detail?id=67_

**Comment by yzmotoxer807** [comment]
Subscript/superscript would be extremely helpful on signal text data and signal names.

**Comment by drom** [comment]
In a ugly way, but it is supported now.
```js
['tspan', 'E=mc',
    ['tspan', {dy:'-5'}, '2'],
    ['tspan', {dy: '5'}, '. '],
    ['tspan', {'font-size':'25'}, 'B '],
    ['tspan', {'text-decoration':'overline'},'over '],
    ['tspan', {'text-decoration':'underline'},'under '],
    ['tspan', {'baseline-shift':'sub'}, 'sub '],
    ['tspan', {'baseline-shift':'super'}, 'super ']
  ]
```
Will give you:

![drome2](https://cloud.githubusercontent.com/assets/511872/10803015/5c3855b6-7d7d-11e5-81f2-ebb9bb2a81c6.png)

**Comment by drom** [comment]
I think we can add several useful classes:
 - [ ] overline
 - [ ] undeline
 - [ ] sub
 - [ ] super
 - [ ] bold
 - [ ] italic

So instead of:

```
['tspan', {'text-decoration': 'overline'}, 'over']
```
one can write:

```
['tspan', {'class': 'overline'}, 'over']
```

**Comment by drom** [comment]
About Wiki markup comparison
http://www.wikimatrix.org/syntax.php

**Comment by drom** [comment]
All SVG text formatting ideas moved to https://github.com/drom/tspan

**Comment by drom** [comment]
Just released v1.2.0
It supports some text formatting tags, including <o>overline</o>
More details here: https://github.com/drom/tspan
